bridge-fungible-token/src/utils.sw

```diff
@@ -63,12 +63,42 @@ fn shift_decimals_left(bn: U256, d: u8) -> Result<U256, BridgeFungibleTokenError
     Result::Ok(bn_clone)
 }
 
+fn shift_decimals_right(bn: U256, d: u8) -> Result<(U256, u32), BridgeFungibleTokenError> {
+    let mut bn_clone = bn;
+    let mut decimals_to_shift: u32 = asm(r1: d) { r1: u32 };
+    let mut r = 0u32;
+
+    // the zero case
+    if (decimals_to_shift == 0u32) {
+        return Result::Ok((bn, 0u32));
+    };
+
+    // the too large case
+    // (there are only 78 decimal digits in a 256bit number)
+    if (decimals_to_shift > 77u32) {
+        return Result::Err(BridgeFungibleTokenError::OverflowError);
+    };
+
+    // math time
+    while (decimals_to_shift > 0u32) {
+        if (decimals_to_shift < 20u32) {
+            let (adjusted, remainder) = bn_div(bn_clone, 10u32.pow(decimals_to_shift));
+            return Result::Ok((adjusted, remainder));
+        } else {
+            let (adjusted, remainder) = bn_div(bn_clone, 10u32.pow(19u32));
+            decimals_to_shift = decimals_to_shift - 19u32;
+            bn_clone -= adjusted;
```

Comment:
We don't need to be subtracting anything, only doing multiple divisions/multiplications in succession.
*note: this also applies to `shift_decimals_left` and turn the `+=` into just `=`

```suggestion
            bn_clone = adjusted;
```